build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/DockerPush.java

```diff
@@ -0,0 +1,114 @@
+package org.jfrog.build.extractor.docker.extractor;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.jfrog.build.api.Build;
+import org.jfrog.build.api.Module;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryBuildInfoClientBuilder;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryClientConfiguration;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryDependenciesClientBuilder;
+import org.jfrog.build.extractor.docker.DockerAgentUtils;
+import org.jfrog.build.extractor.docker.types.DockerImage;
+import org.jfrog.build.extractor.packageManager.PackageManagerExtractor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.jfrog.build.extractor.packageManager.PackageManagerUtils.createArtifactoryClientConfiguration;
+
+public class DockerPush extends PackageManagerExtractor {
+    ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder;
+    ArtifactoryDependenciesClientBuilder dependenciesClientBuilder;
```

Comment:
ArtifactoryDependenciesClientBuilder --> "dependencies" is for downloads only, not uploads.

Reply:
https://github.com/jfrog/build-info/pull/373#discussion_r471093629